src/providers/scicat-provider/dao/scicat-be-dao.ts

```diff
@@ -0,0 +1,115 @@
+import logger from "../../../server/logger";
+import { Configuration, PublishedDataApi, PublishedDataControllerFindAllRequest, PublishedDataControllerFindOneRequest } from "@scicatproject/scicat-sdk-ts-fetch";
+import { AppConfiguration } from "../../../server/app-configuration"
+
+/**
+ * This is the DAO service for Scicat. It uses a mongo connection
+ * to retrieve data.  Database connection parameters are
+ * provided by the credentials file (path defined in .env).
+ */
+export class SciCatBEConnector {
+  public static instance: SciCatBEConnector;
+  public publishedDataApi: PublishedDataApi;
+  public scicatUrl: string;
+  public publishedDataCount: number;
+  private aconf: AppConfiguration;
+
+  private constructor() {
+
+    logger.debug("Setting up scicat client.");
+
+    this.aconf = AppConfiguration.instance;
+
+    this.scicatUrl = this.aconf.scicat_backend_url;
+    if (this.scicatUrl == "") {
+      logger.error("SciCat Backend URL should be defined")
+      throw new Error("SciCat Backend URL should be defined")
+    }
+    logger.info("SciCat url: " + this.scicatUrl);
+
+    if (!this.publishedDataApi) {
+
+      const apiConfig = new Configuration({
+        basePath: this.scicatUrl,
+      });
+
+      this.publishedDataApi = new PublishedDataApi(apiConfig);
+
+      let publishedDataCount = -1;
+      this.publishedDataApi.publishedDataControllerCount()
+        .then( (res) =>{
+          publishedDataCount = res.count;
+        })
+        .catch( (error) => {
+          logger.error("Failed to connect to SciCat :", error.message);
+          throw error;
+        });
+      this.publishedDataCount = publishedDataCount; 
```

Comment:
```suggestion
      this.publishedDataApi.publishedDataControllerCount()
        .then( (res) =>{
          this.publishedDataCount = res.count;
        })
        .catch( (error) => {
          logger.error("Failed to connect to SciCat :", error.message);
          throw error;
        });
```

The local variable is not needed, and `this.publishedDataCount` would always to set to -1 because the last line would be executed before the async callback.

Reply:
Good point. Thank you for catching this.

Reply:
Good catch.